tests/requirements/lint-mypy.txt

```diff
@@ -0,0 +1,2 @@
+mypy
+types-PyYAML
```

Comment:
We should pin the versions too to avoid issues in CI if new checks are added to the latest version of these linters.

Reply:
on linting I would rather prefer to be always on latest version, to keep updating the source code to latest features/rules

Reply:
that's the whole point :)
I'd prefer to find changes in the linter during CI and fix ASAP on the limited piece of code that's affected, than pinning to a version and having to refactor much more code when and if the pinned version will be bumped